data/src/main/java/com/microsoft/azure/kusto/data/ClientImpl.java

```diff
@@ -19,10 +19,21 @@
 
     private AadAuthenticationHelper aadAuthenticationHelper;
     private String clusterUrl;
+    private String clientVersionForTracing;
 
     public ClientImpl(ConnectionStringBuilder csb) throws URISyntaxException {
         clusterUrl = csb.getClusterUrl();
         aadAuthenticationHelper = new AadAuthenticationHelper(csb);
+        if(StringUtils.isNotBlank(csb.getClientVersionForTracing()) ){
+            clientVersionForTracing = "Kusto.Java.Client";
+            String version = Utils.GetPackageVersion();
+            if (StringUtils.isNotBlank(version)) {
+                clientVersionForTracing += ":" + version;
+            }
+        }
+        else{
+            clientVersionForTracing = csb.getClientVersionForTracing();
```

Comment:
if getClientVersionForTracing is blank, don't you want to assign something else?